### PR TITLE
typesctipt

### PR DESCRIPTION
**Pull request by atoshmukhamedov**
rewrite in rewrite in

### CODE REVIEW

File: config/database.ts
Comment (Xayrulloh) [inline config/database.ts:1]: enter to your chatgpt, i've asked to configure mongoose with typescript, we need to change it

File: config/storage.ts
Comment (Xayrulloh) [inline config/storage.ts:1]: format

File: cron/cron.ts
Comment (Xayrulloh) [inline cron/cron.ts:1]: format

File: cron/cron.ts [hunk 1]
```diff
@@ -21,18 +21,18 @@ export async function monthly() {
   await Model.PrayTime.deleteMany()
 
   for (let i = 0; i < regions.length; i++) {
-    const pdf = await axios.get(process.env.TIME_API + regionIds[i] + '/' + currentMonth, {
+    const pdf = await axios.get(String(process.env.TIME_API) + regionIds[i] + '/' + currentMonth, {
       responseType: 'arraybuffer',
     })
-    const pdfData = await pdfParser(pdf.data, { normalizeWhitespace: true })
```
Comment (Xayrulloh) [inline cron/cron.ts:27]: why u deleted {normalizeWhiteSpace: true}

File: cron/cron.ts [hunk 2]
```diff
@@ -52,102 +52,99 @@ export async function monthly() {
...
   // sending message
   const now = new Date()
   const currentDay = now.getDate()
-  const regions = await Model.PrayTime.find({ day: currentDay })
+  const regions = await Model.PrayTime.find({day: currentDay})
 
   // taking hadith
+  let hadith = await Model.Hadith.find({category: {$ne: 'juma'}})
   if (now.getDay() == 5) {
-    var hadith = await Model.Hadith.find({ category: 'juma' })
```
Comment (Xayrulloh) [inline cron/cron.ts:73]: do u've problem with "var"

File: cron/cron.ts [hunk 3]
```diff
@@ -52,102 +52,99 @@ export async function monthly() {
...
-    hadith = hadith[(Math.random() * hadith.length) | 0]
-    var file = new InputFile('./uploads/JumaMuborak.jpg')
-  } else {
-    var hadith = await Model.Hadith.find({ category: { $ne: 'juma' } })
-    hadith = hadith[(Math.random() * hadith.length) | 0]
+    hadith = await Model.Hadith.find({category: 'juma'})
   }
+  const randomHadith = hadith[(Math.random() * hadith.length) | 0]
```
Comment (Xayrulloh) [inline cron/cron.ts:78]: theres no need to declare new variable

File: keyboard/custom.ts
Comment (Xayrulloh) [inline keyboard/custom.ts:1]: format

File: scenes/advertise.ts [hunk 4]
```diff
@@ -17,14 +18,14 @@ scene.wait().on('message:text', async (ctx) => {
   for (const user of users) {
     try {
       await ctx.api.sendMessage(user.userId, ctx.message.text)
-    } catch (error) {
+    } catch (error: any) {
```
Comment (Xayrulloh) [inline scenes/advertise.ts:21]: I think it has a type for Error

File: types/context.ts
Comment (Xayrulloh) [inline types/context.ts:1]: can u explain, what hell is going on in here

**Review by Xayrulloh — CHANGES_REQUESTED** [review]

File: cron/cron.ts (hunk 1 above)
Comment (atoshmukhamedov) [reply cron/cron.ts:27]: there's not an option named normalizeWhiteSpace

File: cron/cron.ts (hunk 2 above)
Comment (atoshmukhamedov) [reply cron/cron.ts:73]: as expected)

File: scenes/advertise.ts (hunk 4 above)
Comment (atoshmukhamedov) [reply scenes/advertise.ts]: non fixable(

File: cron/cron.ts
Comment (atoshmukhamedov) [reply cron/cron.ts:1]: fixed

**Pull request merged**